src/inmanta/protocol/rest/__init__.py

```diff
@@ -131,7 +131,7 @@ def _map_headers(self, arg: str) -> Optional[Any]:
         if opts.header is None:
             return None
 
-        value = self._request_headers[opts.header]
+        value = self._request_headers.get(opts.header)
```

Comment:
The header should not be set in the reply, when it is not set in the request. This implementation will set the reply header to None, when value==None